src/main/java/seedu/address/logic/commands/UntagCommand.java

```diff
@@ -0,0 +1,123 @@
+    public static final String MESSAGE_DUPLICATE_PERSON = "This person already exists in the address book.";
+    public static final String MESSAGE_TAG_NOT_IN_PERSON = "This person does not have the following tags: %s";
+
+    /**
+     * @param index of the person in the filtered person list to edit
+     * @param editPersonDescriptor details to edit the person with
+     */
+    public UntagCommand(Index index, EditPersonDescriptor editPersonDescriptor) {
+        super(index, editPersonDescriptor);
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (getIndex().getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Person personToEdit = lastShownList.get(getIndex().getZeroBased());
+        Person editedPerson = createEditedPerson(personToEdit, getEditPersonDescriptor());
+
+        if (!personToEdit.isSamePerson(editedPerson) && model.hasPerson(editedPerson)) {
+            throw new CommandException(MESSAGE_DUPLICATE_PERSON);
+        }
+
+        model.setPerson(personToEdit, editedPerson);
+        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+        return new CommandResult(String.format(MESSAGE_REMOVE_PERSON_SUCCESS, editedPerson.getName(),
+                getRemovedTags(getEditPersonDescriptor())));
+    }
+
+    /**
+     * Creates and returns a {@code Person} with the details of {@code personToEdit}
+     * edited with {@code editPersonDescriptor}.
+     */
+    private static Person createEditedPerson(Person personToEdit, EditPersonDescriptor editPersonDescriptor)
+            throws CommandException {
+        assert personToEdit != null;
+
+        Name updatedName = editPersonDescriptor.getName().orElse(personToEdit.getName());
+        Phone updatedPhone = editPersonDescriptor.getPhone().orElse(personToEdit.getPhone());
+        Email updatedEmail = editPersonDescriptor.getEmail().orElse(personToEdit.getEmail());
+        Address updatedAddress = editPersonDescriptor.getAddress().orElse(personToEdit.getAddress());
+        Birthday updatedBirthday = editPersonDescriptor.getBirthday().orElse(personToEdit.getBirthday());
+
+        Set<Tag> removedTags = editPersonDescriptor.getTags().orElse(new HashSet<Tag>());
+        Set<Tag> updatedTags = new HashSet<>(personToEdit.getTags());
+        if (!updatedTags.containsAll(removedTags)) {
+            throw new CommandException(String.format(MESSAGE_TAG_NOT_IN_PERSON,
+                    getNotFoundTags(updatedTags, removedTags)));
+        }
+        updatedTags.removeAll(removedTags);
+
+        return new Person(updatedName, updatedPhone, updatedEmail, updatedAddress, updatedTags, updatedBirthday);
+    }
+
+    public static String getNotFoundTags(Set<Tag> originalTags, Set<Tag> removedTags) {
```

Comment:
Could you add docstrings to these methods?
Tracing inline and nested ternary and filters is not easy for me :(

Also can consider pulling out the functions as variables, perhaps 
Filter TagIsInOriginalTags = ...
removedTags.Stream.filter(TagIsInOriginalTags)
then do the uglier tag extraction inside the pulled out variable/filter